source/_patterns/02-components/language-switcher/language-switcher.scss

```diff
@@ -33,19 +34,28 @@
 			z-index: $z-index-rea-header-cmp-language-switcher-li; // * TODO: possibly rework variable naming
 
 			// TODO: This is an enhancement for more languages, but obviously doesn't scale that nicely and would need some rework
-			&:not([aria-selected="true"]) ~ li:not([aria-selected="true"]) {
+			&:not([aria-current="page"], [aria-selected="true"])
+				~ li:not([aria-current="page"], [aria-selected="true"]) {
 				top: to-rem($pxValue: 54 * 2);
 
-				& ~ li:not([aria-selected="true"]) {
+				& ~ li:not([aria-current="page"], [aria-selected="true"]) {
 					top: to-rem($pxValue: 54 * 3);
 
-					& ~ li:not([aria-selected="true"]) {
+					& ~ li:not([aria-current="page"], [aria-selected="true"]) {
 						top: to-rem($pxValue: 54 * 4);
 
-						& ~ li:not([aria-selected="true"]) {
+						&
+							~ li:not(
+								[aria-current="page"],
+								[aria-selected="true"]
+							) {
 							top: to-rem($pxValue: 54 * 5);
 
-							& ~ li:not([aria-selected="true"]) {
+							&
+								~ li:not(
+									[aria-current="page"],
+									[aria-selected="true"]
+								) {
 								top: to-rem($pxValue: 54 * 6);
```

Comment:
this whole block is very deeply nested and so hard to understand. May we change this to a less nested structure?

Reply:
you're right, but I actually don't want to invest too much time into this, as that's an old 1.6 guidelines component. Do you have a quick fix / solution for this?